webrender_api/src/api.rs

```diff
@@ -471,6 +477,24 @@ impl RenderApi {
             .unwrap();
     }
 
+    /// Add/remove/update resources such as images and fonts.
+    ///
+    /// This is similar to update_resources with the addition that it allows updating
+    /// a pipeline's epoch.
+    pub fn update_pipeline_resources(
+        &self,
+        resources: ResourceUpdates,
+        document_id: DocumentId,
+        pipeline_id: PipelineId,
+        epoch: Epoch,
+    ) {
+        self.send(document_id, DocumentMsg::UpdatePipelineResources {
```

Comment:
Not sure I understand the semantics of this. What is the relation of `pipeline_id` and `document_id` with the given resource updates?
If the resources are supposed to be used by this pipeline/document only, we need to reflect it somehow. If not, then it's not clear why would the user want to bump the epoch of an arbitrary pipeline while doing some of the resource updates.

> This is similar to update_resources with the addition that it allows updating a pipeline's epoch.

This begs for a question why it's not just a separate method to bump the pipeline epoch.